### ISSUE TITLE
How me check that cache is correct?

### ISSUE THREAD

**Issue by mefilt** [grab/cocoapods-binary-cache#23]
How me check that cache is correct?
I want building cache when cache invalid.

**Comment by trinhngocthuyen** [comment]
Hi @mefilt,
Let me quickly brief you on how it works.

TLDR: Run the CLI: `bundle exec pod binary prebuild`

### Integration of cache hit/miss
Currently, the cache validity is done internally. A framework cache is called a **cache hit** if the cache is valid to use. Otherwise, it's a **cache miss**.

During the integration:
- _A cache hit_ is integrated using prebuilt frameworks
- _A cache miss_ is integrated as normal using its source code (like integration without this plugin)

When prebuilding the frameworks, by default, those detected as cache miss will be re-built again. To prebuild frameworks, use the command:
```sh
bundle exec pod binary prebuild
```
Kindly check out out the [CLI section](https://github.com/grab/cocoapods-binary-cache#3-cli) for more details.

### Logic of cache miss/hit
There are 2 types of pods: development pods and non-development pods.
- **Non-development pods**: The cache is version-based.
- **Development pods**: The cache validity is based on the checksum of the pod folder.\
\
The state of the cache (used for cache validation) is captured by the `Manifest.lock` in the cache repo and the metadata of each framework when we prebuilt them.

Moreover, for both types, you can add the check on the build settings of a cache during cache validation. A use case is when a non-development pod has no change in its source code, but we change it from dynamic to static. You can check out the `validate_prebuilt_settings ` option in [this doc](https://github.com/grab/cocoapods-binary-cache/blob/master/docs/configure_cocoapods_binary_cache.md).
...

Hope this answers your question
